Implementing Shor's algorithm using the Qrisp framework on Cat Qubits.ipynb

```diff
@@ -0,0 +1,770 @@
+    "\n",
+    "Shor's algorithm requires considerably low error rates to give proper results - we will talk about numbers in a following section, when we will do a sketch of the resource estimation of the algorithm - and therefore error correction is needed. So we tought - wait a second, why don't we use cat qubits then, which are the qubits of the future?\n",
+    "\n",
+    "### Why Qrisp?\n",
+    "\n",
+    "Qrisp is a high-level programming language for creating and compiling quantum algorithms which features numerous characteristics to smooth the workflow when writing and executing a quantum algorithm.\n",
+    "If you want to get familiar with the Qrisp framework, have a look at the dedicated [series of tutorials](https://qrisp.eu/general/tutorial/index.html) available on Qrisp's website.\n",
+    "\n",
+    "Qrisp provides compatibility with Qiskit's framework and it is possible to run Qrisp code directly on the Qiskit's backend making use of the `VirtualQiskitBackend`. This allows us to instantiate the emulator using the [Alice & Bob Qiskit Provider]() as backend and run the code on a cat qubits architechture.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "3a32e67d-f628-4f47-a945-b44a342cd6ac",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qrisp import *\n",
+    "from qiskit import *\n",
+    "from qrisp.interface import VirtualQiskitBackend\n",
+    "from qiskit_alice_bob_provider.local.provider import AliceBobLocalProvider\n",
+    "\n",
+    "provider = AliceBobLocalProvider()\n",
+    "backend = provider.get_backend('EMU:40Q:LOGICAL_TARGET')\n",
+    "vrtl_qasm_sim = VirtualQiskitBackend(backend)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "963c9533",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import math\n",
+    "import random\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c38e52c5",
+   "metadata": {},
+   "source": [
+    "We would like to perform the trivial task of prime factoring the number $15$. Everybody knows that $5\\times3 = 15$, but sometimes is better to double-check - or triple-check, as in this case.\n",
+    "\n",
+    "Why don't we start by comparing a classical brute-force period finding algorithm with its quantum counterpart and analyse their performances?\n",
+    "The implementation of this version of the Shor's algorithm is inspired by Qrisp'official tutorial."
```

Comment:
I don't know if this is relevant to say

Reply:
This was mentioned by Laurent, since its initial version was basically a copy of  their tutorial. I kept one function from their tutorial, so maybe it is still worth mentioning it

Reply:
Cool, let's keep it then!

Reply:
Changed my mind, removed